### ISSUE TITLE
Consider using https://github.com/Microsoft/wil to make code cleaner

### ISSUE THREAD

**Issue by Alan-Jowett** [microsoft/ebpf-for-windows#664]
Consider using [wil](https://github.com/Microsoft/wil) to make code cleaner. The WIL library simplifies common coding patterns and can make our code a lot cleaner.

**Comment by dthaler** [comment]
WIL says it's C++ so is only applicable to user-mode code in this repo.

**Comment by Alan-Jowett** [comment]
Yes, this would be applicable to the ebpfapi layer and the ebpfsvc.
